source/Plugins.h

```diff
@@ -0,0 +1,70 @@
+/* Plugins.h
+Copyright (c) 2022 by Sam Gleske (samrocketman on GitHub)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef PLUGINS_H_
+#define PLUGINS_H_
+
+#include "Set.h"
+
+#include <string>
+
+
+
+// Represents information about a single plugin.
+struct Plugin {
```

Comment:
Can we make this a class in a separate file?
That would be more how we generally structure our things.

Reply:
Not sure it makes sense to break this out.  Its intent is to track state and the only place it would ever be used is this Plugins file.  Breaking out into classes seems like it would be better intended if code is meant to for reuse at some point.  I can't imagine this getting reused.

Reply:
so in that case, could you move this struct into the Plugins class, to match how we normally do it

Reply:
@RisingLeaf I can still open a follow up PR if you want

Reply:
It seems like warp core thought this is fine, so I do not think a follow up is necessary

Reply:
I was going to change it but I forgot by the time I got to the end of the changes and tested it.